oxenstored/process.ml

```diff
@@ -341,6 +341,63 @@ let do_directory con t _domains _cons data =
   else
     ""
 
+let do_directory_part con t _domains _cons data =
+  (* Call only available to Dom0 *)
+  if not (Connection.is_dom0 con) then
+    raise Domain_not_match ;
+
+  let directory_cache = Option.get con.Connection.directory_cache in
+
+  let split_two_args data con =
+    let args = split (Some 3) '\000' data in
+    match args with
+    | path :: offset :: _ ->
+        (Store.Path.create path (Connection.get_path con), int_of_string offset)
+    | _ ->
+        raise Invalid_Cmd_Args
+  in
+
+  (* First arg is node name. Second arg is childlist offset. *)
+  let path, offset = split_two_args data con in
+
+  let generation, children =
+    Transaction.ls_partial t (Connection.get_perm con) path directory_cache
+  in
+
+  let generation_s = Int64.to_string generation ^ "\000" in
+  let genlen = String.length generation_s in
+  let children_length = String.length children in
+
+  (* Offset behind list: just return a list with an empty string. *)
+  if offset >= children_length then
+    generation_s ^ "\000"
```

Comment:
Is this adding more zeros after the existing zero?

Reply:
Yes, this signifies the last packet to differentiate it from all the intermediate ones